doc/rationale.md

```diff
@@ -0,0 +1,31 @@
+# Ignition rationale
+
+Ignition is a distribution-agnostic provisioning utility. When a machine boots for the first time, Ignition will manipulate it in various ways to set up the machine for operations.
+
+There are several principles that have guided Ignition's design. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool, not a configuration management tool. Ignition encourages immutable infrastructure, wherein if a user wishes to modify a machine, they should discard the old node and re-provision the machine. This keeps the user's machines in a well known state with relatively simple tooling.
+
+To this end, Ignition runs on the first boot of a machine. Being a general provisioning utility however, there's nothing stopping a user from using Ignition to set up some configuration management tool. This may even make a lot of sense depending on the organization and requirements the user is trying to meet.
+
+From the user's perspective this first boot is not special in any way. The system is fully provisioned and ready for use once fully booted.
+
+## Ignition produces the machine specified or no machine at all
+
+Ignition does what it needs to make the system match the state described in the Ignition config. If for any reason Ignition cannot deliver the exact machine that the config asked for, Ignition prevents the machine from booting successfully.
+
+For example, if the user wanted to fetch the document hosted at `https://example.com/foo.conf` and write it to disk, Ignition would prevent the machine from booting in the event that it is unable to resolve the given URL.
+
+## Ignition configs are declarative
+
+Ignition configs describe the state of a system. Ignition configs do _not_ list a series of steps that Ignition should take.
+
+To this end, Ignition configs do not allow users to provide arbitrary logic (including scripts for Ignition to run). Users describe which filesystems they want to exist, what files need to be created, what users should exist, and more. Any further customization needs to be using systemd services, created by Ignition.
+
+## Ignition configs should not be written by hand
+
+Ignition configs are not intended to be human friendly and users should not write Ignition configs by hand. Ignition configs were designed to be human readable, but difficult to write. This was done to discourage users from attempting to write configs by hand.
```

Comment:
Ignition configs were designed to be human readable, but difficult to write, to discourage users from attempting to write configs by hand. 

Use a separate tool to generate Ignition configs. For Container Linux, files called ...